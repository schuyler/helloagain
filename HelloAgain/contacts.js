'use strict';

import { Friends } from './models';
let AddressBook = require('react-native-contacts');

export function loadAllContacts() {
    // Load all the contacts from the system addressbook, and within that list,
    // substitute any contacts we already have in our database (and merging any
    // changes from the addressbook). We have to wait until the Contacts model
    // has loaded fully. Returns a Promise that is resolved when everything is
    // loaded.
    return new Promise((resolve, reject) => {
      Friends.hasLoaded.then(() => {
        AddressBook.getAll((err, contacts) => {
          if (err) {
            console.log("Reading addressbook failed:", err);
            reject(err);
            return;
          }
          // Update contacts from the addressbook with whatever we have stored in
          // our database.
          for (var i = 0; i < contacts.length; i++) {
            var storedContact = Friends.load(contacts[i].recordID);
            if (storedContact) {
              Object.assign(storedContact, contacts[i]);
              if (storedContact.isSelected) { // only write back changes to selected contacts
                Friends.save(storedContact);
              }
              contacts[i] = storedContact;
            }
          }
          resolve(contacts);
        });
      });
    });
};
