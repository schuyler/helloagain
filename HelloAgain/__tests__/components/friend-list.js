'use strict'

import React from 'react';
import renderer from 'react-test-renderer';

import { CONTACT_FIXTURE } from "../fixtures/contacts"
import FriendList from "../../components/friend-list"

it('renders a list of friends', () => {
  expect(renderer.create(
    <FriendList items={CONTACT_FIXTURE} />
  )).toMatchSnapshot();
})
