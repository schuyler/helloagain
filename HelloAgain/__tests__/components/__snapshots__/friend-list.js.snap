exports[`test renders a list of friends 1`] = `
<View
  style={
    Object {
      "paddingTop": 64,
    }
  }>
  <ScrollView
    dataSource={
      ListViewDataSource {
        "items": 17,
      }
    }
    initialListSize={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={1000}
    onKeyboardDidHide={undefined}
    onKeyboardDidShow={undefined}
    onKeyboardWillHide={undefined}
    onKeyboardWillShow={undefined}
    onLayout={[Function]}
    onScroll={[Function]}
    pageSize={1}
    removeClippedSubviews={true}
    renderRow={[Function]}
    scrollEventThrottle={50}
    scrollRenderAheadDistance={1000}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#ffffff",
        "paddingTop": 20,
      }
    }>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#F5FCFF",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-around",
            "padding": 5,
          }
        }>
        <Image
          source={Object {}}
          style={undefined} />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "flex": 15,
              "fontSize": 20,
              "marginBottom": 5,
              "textAlign": "left",
            }
          }>
          Nicolaus
           
          Copernicus
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "flex": 1,
              "fontSize": 15,
              "marginBottom": 5,
              "marginRight": 5,
              "textAlign": "right",
            }
          }>
          ???
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#F5FCFF",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-around",
            "padding": 5,
          }
        }>
        <Image
          source={Object {}}
          style={undefined} />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "flex": 15,
              "fontSize": 20,
              "marginBottom": 5,
              "textAlign": "left",
            }
          }>
          Tycho
           
          Brahe
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "flex": 1,
              "fontSize": 15,
              "marginBottom": 5,
              "marginRight": 5,
              "textAlign": "right",
            }
          }>
          ???
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#F5FCFF",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-around",
            "padding": 5,
          }
        }>
        <Image
          source={Object {}}
          style={undefined} />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "flex": 15,
              "fontSize": 20,
              "marginBottom": 5,
              "textAlign": "left",
            }
          }>
          Clyde
           
          Tombaugh
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "flex": 1,
              "fontSize": 15,
              "marginBottom": 5,
              "marginRight": 5,
              "textAlign": "right",
            }
          }>
          ???
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#F5FCFF",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-around",
            "padding": 5,
          }
        }>
        <Image
          source={Object {}}
          style={undefined} />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "flex": 15,
              "fontSize": 20,
              "marginBottom": 5,
              "textAlign": "left",
            }
          }>
          William
           
          Herschel
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "flex": 1,
              "fontSize": 15,
              "marginBottom": 5,
              "marginRight": 5,
              "textAlign": "right",
            }
          }>
          ???
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#F5FCFF",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-around",
            "padding": 5,
          }
        }>
        <Image
          source={Object {}}
          style={undefined} />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "flex": 15,
              "fontSize": 20,
              "marginBottom": 5,
              "textAlign": "left",
            }
          }>
          John
           
          Herschel
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "flex": 1,
              "fontSize": 15,
              "marginBottom": 5,
              "marginRight": 5,
              "textAlign": "right",
            }
          }>
          ???
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#F5FCFF",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-around",
            "padding": 5,
          }
        }>
        <Image
          source={Object {}}
          style={undefined} />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "flex": 15,
              "fontSize": 20,
              "marginBottom": 5,
              "textAlign": "left",
            }
          }>
          Charles
           
          Messier
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "flex": 1,
              "fontSize": 15,
              "marginBottom": 5,
              "marginRight": 5,
              "textAlign": "right",
            }
          }>
          ???
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#F5FCFF",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-around",
            "padding": 5,
          }
        }>
        <Image
          source={Object {}}
          style={undefined} />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "flex": 15,
              "fontSize": 20,
              "marginBottom": 5,
              "textAlign": "left",
            }
          }>
          Henrietta
           
          Leavitt
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "flex": 1,
              "fontSize": 15,
              "marginBottom": 5,
              "marginRight": 5,
              "textAlign": "right",
            }
          }>
          ???
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#F5FCFF",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-around",
            "padding": 5,
          }
        }>
        <Image
          source={Object {}}
          style={undefined} />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "flex": 15,
              "fontSize": 20,
              "marginBottom": 5,
              "textAlign": "left",
            }
          }>
          Jérôme
           
          Lalande
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "flex": 1,
              "fontSize": 15,
              "marginBottom": 5,
              "marginRight": 5,
              "textAlign": "right",
            }
          }>
          ???
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#F5FCFF",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-around",
            "padding": 5,
          }
        }>
        <Image
          source={Object {}}
          style={undefined} />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "flex": 15,
              "fontSize": 20,
              "marginBottom": 5,
              "textAlign": "left",
            }
          }>
          Ejnar
           
          Herzsprung
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "flex": 1,
              "fontSize": 15,
              "marginBottom": 5,
              "marginRight": 5,
              "textAlign": "right",
            }
          }>
          ???
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#F5FCFF",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-around",
            "padding": 5,
          }
        }>
        <Image
          source={Object {}}
          style={undefined} />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "flex": 15,
              "fontSize": 20,
              "marginBottom": 5,
              "textAlign": "left",
            }
          }>
          Edwin
           
          Hubble
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "flex": 1,
              "fontSize": 15,
              "marginBottom": 5,
              "marginRight": 5,
              "textAlign": "right",
            }
          }>
          ???
        </Text>
      </View>
    </View>
  </ScrollView>
</View>
`;
